workflow/controller/operator.go

```diff
@@ -913,7 +913,8 @@ func (woc *wfOperationCtx) podReconciliation() error {
 			// See processNodeRetries for more details.
 			node.FinishedAt = metav1.Time{Time: time.Now().UTC()}
 			woc.wf.Status.Nodes[nodeID] = node
-			woc.log.Warnf("pod %s deleted", nodeID)
+			woc.log.WithField("displayName", node.DisplayName).WithField("templateName", node.TemplateName).
```

Comment:
why not log this is in `markNodePhase`?

Reply:
This is the single point where Pod updates enter the controller. `markNodePhase` may be used for other errors, but the intention of this PR is to enhance the logs related to direct Pod updates